File: solutions/fable-1/fable/javascript/rle.js
const input = require('fs').readFileSync(0, 'utf8');
const s = input.split(/\r?\n/)[0] || '';
let out = '';
let i = 0;
while (i < s.length) {
  let j = i;
  while (j < s.length && s[j] === s[i]) j++;
  out += s[i] + (j - i);
  i = j;
}
process.stdout.write(out + '\n');

File: solutions/fable-1/fable/rust/rle.rs
use std::io::{self, Read, Write};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let line = input.lines().next().unwrap_or("");
    let bytes = line.trim_end_matches(['\r', '\n']).as_bytes();

    let mut out = String::new();
    let mut i = 0;
    while i < bytes.len() {
        let c = bytes[i];
        let mut j = i;
        while j < bytes.len() && bytes[j] == c {
            j += 1;
        }
        out.push(c as char);
        out.push_str(&(j - i).to_string());
        i = j;
    }
    out.push('\n');

    io::stdout().write_all(out.as_bytes()).unwrap();
}
